robot-db/src/main/java/domain/Book.java

```diff
@@ -93,4 +96,35 @@ public String toString () {
                 .append(description)
                 .append("}").toString();
     }
-}
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17,31)
+                .append(title)
+                .append(newPrice)
+                .append(oldPrice)
+                .append(currency)
+                .append(year)
+                .append(link)
+                .append(description)
+                .append(printHouse)
+                .append(authors)
+                .append(category)
+                .toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object bookObj) {
+        if (this == bookObj)
+            return true;
+        if (bookObj == null || getClass() != bookObj.getClass())
+            return false;
+        Book book = (Book)bookObj;
+        return title.equals(book.title)
+                && abs(newPrice - book.newPrice) < 1e-10 && abs(oldPrice - book.oldPrice) < 1e-10
```

Comment:
No, everything is fine :) This approach is much safer because of numerical representation of float numbers -> IEEE 754.
Of course, I maybe should change epsilon to lower value, but when weconsider prices it should be ok.